src/renderer/errorDialog/js/errorDialog.js

```diff
@@ -0,0 +1,237 @@
+/**
+ * The error dialog
+ *
+ * Represents an error dialog, that can be closed by the user.
+ * Copyright 2017 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ * The research leading to these results has received funding from the European Union's
+ * Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+/* global fluid */
+
+"use strict";
+(function (fluid) {
+    var ipcRenderer = require("electron").ipcRenderer;
+
+    var gpii = fluid.registerNamespace("gpii");
+
+
+    /**
+     * A component that handles connection and communication with the Main process.
+     * It supplies interface (through invokers) for communication in direction to
+     * and events for data coming from the Main process.
+     */
+    fluid.defaults("gpii.errorDialog.channel", {
+        gradeNames: ["fluid.component"],
+
+        events: {
+            onConfigReceived: null
+        },
+
+        listeners: {
+            "onCreate.registerChannel": {
+                funcName: "gpii.errorDialog.channel.register",
+                args: "{that}.events"
+            }
+        },
+
+        invokers: {
+            notify: {
+                funcName: "gpii.errorDialog.channel.notifyChannel"
+            }
+        }
+    });
+
+
+    /**
+     * Sends a message to the main process.
+     * @param {...Any} The channel to be notified and the parameters to be passed
+     * with the message.
+     */
+    gpii.errorDialog.channel.notifyChannel = function () {
+        ipcRenderer.send.apply(null, arguments);
+    };
+
+    /**
+     * Registers for events from the Main process.
+     * @param events {Object} Events map.
+     */
+    gpii.errorDialog.channel.register = function (events) {
+        ipcRenderer.on("onErrorUpdate", function (event, config) {
+            events.onConfigReceived.fire(config);
+        });
+    };
+
+    fluid.defaults("gpii.errorDialog.button", {
+        gradeNames: "gpii.psp.widgets.button",
+        model: {
+            label: null
+        },
+
+        invokers: {
+            onClick: {
+                func: "{errorDialog}.events.onButtonClicked.fire",
+                /* Simply identify a button by its label */
+                args: "{that}.model.label"
+            }
+        },
+        modelListeners: {
+            label: {
+                this: "{that}.container",
+                method: "toggle",
+                args: "@expand:fluid.isValue({change}.value)"
+            }
+        }
+    });
+
+    /**
+     * Responsible for the visualisation of the error dialog.
+     */
+    fluid.defaults("gpii.errorDialog", {
+        gradeNames: ["fluid.viewComponent", "gpii.psp.heightObservable"],
```

Comment:
Out of interest, what is the cause of the height change (during rendering/loading etc.) that requires this to be observed? The explanation should be added to the comments here

Reply:
Added a comment for the grade. Basically, this is necessary because the title, subhead and the actual description of the error are unknown and can have arbitrary length. This on the other hand may necessitate more vertical space than provided in the initial configuration of the `BrowserWindow`. Knowing the exact height makes it possible to adjust the dimensions of the `BrowserWindow` appropriately while not taking any more space than needed.